datafusion/physical-expr/src/intervals/utils.rs

```diff
@@ -0,0 +1,192 @@
+            | &DataType::UInt64
+            | &DataType::UInt32
+            | &DataType::UInt16
+            | &DataType::UInt8
+            | &DataType::Float64
+            | &DataType::Float32
+    )
+}
+
+/// Converts an [`Interval`] of time intervals to one of `Duration`s, if applicable. Otherwise, returns [`None`].
+pub fn convert_interval_type_to_duration(interval: &Interval) -> Option<Interval> {
+    if let (Some(lower), Some(upper)) = (
+        convert_interval_bound_to_duration(&interval.lower),
+        convert_interval_bound_to_duration(&interval.upper),
+    ) {
+        Some(Interval::new(lower, upper))
+    } else {
+        None
+    }
+}
+
+/// Converts an [`IntervalBound`] containing a time interval to one containing a `Duration`, if applicable. Otherwise, returns [`None`].
+fn convert_interval_bound_to_duration(
+    interval_bound: &IntervalBound,
+) -> Option<IntervalBound> {
+    match interval_bound.value {
+        ScalarValue::IntervalMonthDayNano(Some(mdn)) => {
+            interval_mdn_to_duration_ns(&mdn).ok().map(|duration| {
+                IntervalBound::new(
+                    ScalarValue::DurationNanosecond(Some(duration)),
+                    interval_bound.open,
+                )
+            })
+        }
+        ScalarValue::IntervalDayTime(Some(dt)) => {
+            interval_dt_to_duration_ms(&dt).ok().map(|duration| {
+                IntervalBound::new(
+                    ScalarValue::DurationMillisecond(Some(duration)),
+                    interval_bound.open,
+                )
+            })
+        }
+        _ => None,
+    }
+}
+
+/// Converts an [`Interval`] of `Duration`s to one of time intervals, if applicable. Otherwise, returns [`None`].
+pub fn convert_duration_type_to_interval(interval: &Interval) -> Option<Interval> {
+    if let (Some(lower), Some(upper)) = (
+        convert_duration_bound_to_interval(&interval.lower),
+        convert_duration_bound_to_interval(&interval.upper),
+    ) {
+        Some(Interval::new(lower, upper))
+    } else {
+        None
+    }
+}
+
+/// Converts an [`IntervalBound`] containing a `Duration` to one containing a time interval, if applicable. Otherwise, returns [`None`].
+fn convert_duration_bound_to_interval(
+    interval_bound: &IntervalBound,
+) -> Option<IntervalBound> {
+    match interval_bound.value {
+        ScalarValue::DurationNanosecond(Some(duration)) => Some(IntervalBound::new(
+            ScalarValue::new_interval_mdn(0, 0, duration),
+            interval_bound.open,
+        )),
+        ScalarValue::DurationMillisecond(Some(duration)) => Some(IntervalBound::new(
+            ScalarValue::new_interval_dt(0, duration as i32),
+            interval_bound.open,
+        )),
+        _ => None,
+    }
+}
+
+/// Unless the number of [`ScalarValue::IntervalMonthDayNano`] has a non-zero month or day field,
+/// returns the nanoseconds part as it will construct a [`ScalarValue::DurationNanosecond`]. Otherwise, returns an error.
```

Comment:
```suggestion
/// If both the month and day fields of [`ScalarValue::IntervalMonthDayNano`] are zero, this function returns the nanoseconds part. 
/// Otherwise, it returns an error.
```